orttraining/orttraining/python/training/onnxblock/model_accessor.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# global_model.py
+
+from contextlib import contextmanager
+
+
+class ModelAccessor:
+    """This class stores the onnx model that is manipulated by the onnx blocks."""
+
+    def __init__(self, model):
+        self.model = model
+        self.eval_model = None
+
+
+# This variable resides in the global namespace.
+# Different methods can access this global model and manipulate it.
+# Its construction and destruction is managed by the onnx_model contextmanager
+global_accessor = None
+
+
+@contextmanager
+def onnx_model(model=None):
```

Comment:
did not see who called it. where is it used?

Reply:
So, I updated the API (so as not to make copies of onnx models) to accept a global onnx model which will be manipulated. Please see the PR description for actual usage. In essence, users will use the API as such:

```py:
model = MyOnnxBlockMode()

# Manipulate the onnx model using the base onnx model as the base
with onnxblock.onnx_model(base_onnx_model) as accessor:
    model_output_names = model()
    eval_model = accessor.eval_model

# Print the model output names
print(model_output_names)
```